framework/db/Query.php

```diff
@@ -590,12 +590,47 @@ public function addSelect($columns)
         if ($this->select === null) {
             $this->select = $columns;
         } else {
-            $this->select = array_merge($this->select, $columns);
+            $this->select = array_merge($this->select, $this->removeDuplicatedColumns($columns));
         }
 
         return $this;
     }
 
+    /**
+     * Removes duplicated columns from given columns definitions.
+     * Columns to be removed:
+     * - if column definition already present in SELECT part with same alias
+     * - if column definition without alias already present in SELECT part without alias too
+     * @param array $columns the columns to be merged to the select.
+     */
+    protected function removeDuplicatedColumns($columns)
```

Comment:
Since the function returns and doesn't affect `$columns`, I'd name it `getUniqueColumns()`.